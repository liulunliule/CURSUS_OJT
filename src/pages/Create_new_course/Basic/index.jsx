import React from "react";
import "./index.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFaceSmile } from "@fortawesome/free-regular-svg-icons";
import ReactQuill from "react-quill";
import TextArea from "antd/es/input/TextArea";
import { Select } from "antd";

const courseLevelOptions = [
  { value: "Beginner", label: "Beginner" },
  { value: "Intermediate", label: "Intermediate" },
  { value: "Advanced", label: "Advanced" },
];

const audioLanguageOptions = [
  { value: "English", label: "English" },
  { value: "Spanish", label: "Spanish" },
  { value: "French", label: "French" },
  { value: "German", label: "German" },
];

const closeCaptionOptions = [
  { value: "English", label: "English" },
  { value: "Spanish", label: "Spanish" },
  { value: "French", label: "French" },
  { value: "German", label: "German" },
];

const courseCategoryOptions = [
  { value: "Web Development", label: "Web Development" },
  { value: "Data Science", label: "Data Science" },
  { value: "Machine Learning", label: "Machine Learning" },
  { value: "Artificial Intelligence", label: "Artificial Intelligence" },
];

function Basic({ basicInfo, setBasicInfo }) {
  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setBasicInfo((prev) => ({ ...prev, [name]: value }));
  };

  const handleSelectChange = (value, name) => {
    setBasicInfo((prev) => ({ ...prev, [name]: value }));
  };

  return (
    <div className="create_course_basic">
      <div className="basic_title">
        <FontAwesomeIcon icon={faFaceSmile} className="basic_icon" />
        Basic information
      </div>

      <div className="basic_main_content">
        <div className="basic_main_course-title">
          <label htmlFor="titlecourse">Course title*</label>
          <form action="" className="course_title-input">
            <input
              type="text"
              name="titlecourse"
              value={basicInfo.titlecourse}
              onChange={handleInputChange}
              placeholder="Course title here"
            />
            <span>60</span>
          </form>
          <p className="course_title-desc">
            (Please make this a maximum of 100 characters and unique.)
          </p>
        </div>

        <div className="basic_main_short-description">
          <label htmlFor="ShortDescription">Short Description*</label>
          <form action="" className="short-description-input">
            <textarea
              name="ShortDescription"
              value={basicInfo.ShortDescription}
              onChange={handleInputChange}
              placeholder="Item description here..."
            />
          </form>
          <p className="course_title-desc">220 words</p>
        </div>

        <div className="basic_main_course-description">
          <label htmlFor="Description">Course Description*</label>
          <ReactQuill
            value={basicInfo.Description}
            onChange={(value) =>
              setBasicInfo((prev) => ({ ...prev, Description: value }))
            }
            theme="snow"
            style={{ height: "150px" }}
          />
        </div>

        <div className="basic_main_requirement">
          <div className="requirement_left" style={{ width: "50%" }}>
            <label htmlFor="Whatlearn">
              What will students learn in your course?*
            </label>
            <TextArea
              name="Whatlearn"
              value={basicInfo.Whatlearn}
              onChange={(e) => handleInputChange(e)}
              rows={5}
            />
            <p className="course_title-desc">
              Student will gain this skills, knowledge after completing this
              course. (One per line).
            </p>
          </div>
          <div className="requirement_right" style={{ width: "50%" }}>
            <label htmlFor="requirement">Requirement*</label>
            <TextArea rows={5} />
            <p className="course_title-desc">
              What knowledge, technology, tools required by users to start this
              course. (One per line).
            </p>
          </div>
        </div>

        <div className="basic_main_select">
          <div className="basic_main_select-item">
            <label htmlFor="courseLevel">Course Level*</label>
            <Select
              style={{
                width: "100%",
                height: "40px",
              }}
              placeholder="Nothing selected"
              options={courseLevelOptions}
              onChange={(value) => handleSelectChange(value, "courseLevel")}
            />
          </div>
          <div className="basic_main_select-item">
            <label htmlFor="audioLanguage">Audio Language*</label>
            <Select
              style={{
                width: "100%",
                height: "40px",
              }}
              placeholder="Select audio"
              options={audioLanguageOptions}
              onChange={(value) => handleSelectChange(value, "audioLanguage")}
            />
          </div>
          <div className="basic_main_select-item">
            <label htmlFor="closeCaption">Close Caption*</label>
            <Select
              style={{
                width: "100%",
                height: "40px",
              }}
              placeholder="Select caption"
              options={closeCaptionOptions}
              onChange={(value) => handleSelectChange(value, "closeCaption")}
            />
          </div>
          <div className="basic_main_select-item">
            <label htmlFor="courseCategory">Course Category*</label>
            <Select
              style={{
                width: "100%",
                height: "40px",
              }}
              defaultValue="Web Development"
              options={courseCategoryOptions}
              onChange={(value) => handleSelectChange(value, "courseCategory")}
            />
          </div>
        </div>
      </div>
    </div>
  );
}

export default Basic;
