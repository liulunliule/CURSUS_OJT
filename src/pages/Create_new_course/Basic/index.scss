.create_course_basic {
    .basic_title {
        padding: 19px 0;
        border-top: 1px solid #efefef;
        border-bottom: 1px solid #efefef;
        display: flex;
        font-size: 18px;
        font-family: "Roboto", sans-serif;
        font-weight: 500;
        gap: 15px;

        .basic_icon {
            font-size: 20px;
        }
    }

    .basic_main_content {
        margin-top: 31px;
        padding: 30px;
        background-color: #fff;
    }

    .basic_main_select-item,
    .requirement_left,
    .requirement_right,
    .basic_main_course-description,
    .basic_main_short-description,
    .basic_main_course-title {
        label {
            font-size: 14px;
            font-weight: 500;
            margin-bottom: 10px;
            color: #333;
        }

        .course_title-input {
            display: flex;
            align-items: center;
            width: 100%;
            height: 40px;
            border: 1px solid #efefef;

            input {
                width: 100%;
                height: 100%;
                padding-left: 1em;
                padding-right: 2em;
                font-size: 14px;
                font-weight: 400;
                border: none;
                outline: none;

                &::placeholder {
                    color: #c7c7c7;
                }
            }

            span {
                background: #ed2a26;
                color: #fff;
                font-weight: 500;
                right: 3px;
                border-radius: 3px;
                height: 34px;
                top: 3px;
                width: 34px;
                font-size: 14px;
                padding: 8px 9px;
            }
        }

        .course_title-desc {
            margin-top: 5px;
            color: #686f7a;
            font-size: 12px;
        }
    }

    .short-description-input {
        height: 130px;
        border: 1px solid #efefef;

        textarea {
            width: 100%;
            height: 100%;
            border: none;
            outline: none;
            padding-left: 16px;
            padding-top: 10px;
            font-size: 14px;
            font-weight: 400;

            &::placeholder {
                color: #c7c7c7;
            }
        }
    }

    .basic_main_requirement {
        margin-top: 71px;
        display: flex;
        gap: 21px;
    }

    .basic_main_select {
        display: grid;
        grid-template-columns: repeat(2, 1fr);
        gap: 30px 21px;
    }
}
