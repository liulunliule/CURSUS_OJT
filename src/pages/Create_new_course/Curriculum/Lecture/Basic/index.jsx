import React from "react";
import "./index.scss";
import { useState } from "react";
import ReactQuill from "react-quill";
import "react-quill/dist/quill.snow.css";
import { Switch } from "antd";
function Add_lecture_basic() {
    return (
        <div className="add_lecture_basic">
            <div className="add_lecture_basic-top">
                <div className="add_lecture_basic-title">Lecture Title*</div>
                <input
                    type="text"
                    name=""
                    id=""
                    className="add_lecture_basic-input"
                    placeholder="Title here"
                />
            </div>
            <div className="add_lecture_basic-bot">
                <div className="add_lecture_basic-desc">Description*</div>
                <ReactQuill theme="snow" style={{ height: "150px" }} />
            </div>
            <div className="add_lecture_basic-switch">
                Free Preview <Switch />
            </div>
            <div className="add_lecture_basic-btn-wrap">
                <button className="add_lecture_basic-btn close">Close</button>
                <button className="add_lecture_basic-btn">Add lecture</button>
            </div>
        </div>
    );
}

export default Add_lecture_basic;
