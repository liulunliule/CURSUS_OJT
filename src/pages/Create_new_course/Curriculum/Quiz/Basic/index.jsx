import React from "react";
import "./index.scss";
import ReactQuill from "react-quill";
import "react-quill/dist/quill.snow.css";
import { Switch } from "antd";
function Add_quiz_basic() {
    return (
        <div className="add_lecture_basic">
            <div className="add_lecture_basic-top">
                <div className="add_lecture_basic-title">Lecture Title*</div>
                <input
                    type="text"
                    name=""
                    id=""
                    className="add_lecture_basic-input"
                    placeholder="Title here"
                />
            </div>
            <div className="add_lecture_basic-bot">
                <div className="add_lecture_basic-desc">Description*</div>
                <ReactQuill theme="snow" style={{ height: "150px" }} />
            </div>
            <div className="add_lecture_basic-switch"></div>
            <div className="add_lecture_basic-btn-wrap">
                <button className="add_lecture_basic-btn close">Close</button>
                <button className="add_lecture_basic-btn">Add lecture</button>
            </div>
        </div>
    );
}

export default Add_quiz_basic;
