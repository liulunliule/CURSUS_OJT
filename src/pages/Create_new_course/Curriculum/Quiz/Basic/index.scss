.add_lecture_basic {
    padding: 29px 14px 0;

    .add_lecture_basic-desc,
    .add_lecture_basic-title {
        font-size: 14px;
        font-weight: 500;
        font-family: "Roboto", sans-serif;
        margin-bottom: 10px;
        color: #333;
    }

    .add_lecture_basic-input {
        font-size: 14px;
        font-weight: 400;
        width: 100%;
        border: 1px solid #efefef;
        height: 40px;
        padding: 0 16px;
        background: #fff;
        border-radius: 3px;
        box-shadow: inset 0px 1px 4px rgb(0 0 0 / 2%);
        outline: none;
    }

    .add_lecture_basic-bot {
        margin-top: 30px;
    }
    .add_lecture_basic-switch {
        display: flex;
        align-items: center;
        gap: 10px;
        margin-top: 60px;
    }
}

.add_lecture_basic-btn-wrap {
    display: flex;
    align-items: center;
    gap: 7px;
    justify-content: flex-end;
    margin-top: 28px;
    padding-top: 15px;
    border-top: 1px solid #dee2e6;
}

.add_lecture_basic-btn {
    height: 40px;
    padding: 0 20px;
    font-size: 14px;
    color: #fff;
    border: 0;
    background: #ed2a26;
    border-radius: 3px;
    font-weight: 500;

    &.close {
        color: #91699c;
        background: #f3f3f3;
    }
}

.ant-switch.ant-switch-checked:hover:not(.ant-switch-disabled) {
    background: #ed2a26;
}
