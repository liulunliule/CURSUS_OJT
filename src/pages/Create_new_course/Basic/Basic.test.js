import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event'; // Import user-event
import Basic from '.';

// Mock ReactQuill component to avoid unnecessary complexity in tests
jest.mock('react-quill', () => (props) => (
  <textarea {...props} />
));

const basicInfo = {
  titlecourse: '',
  ShortDescription: '',
  Description: '',
  Whatlearn: '',
  audioLanguage: '',
  closeCaption: '',
  courseCategory: '',
  courseLevel: '',
};

const mockSetBasicInfo = jest.fn();

// test('handles titlecourse input change', () => {
//   render(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />);

//   fireEvent.change(screen.getByPlaceholderText('Course title here'), {
//     target: { name: 'titlecourse', value: 'New Course Title' },
//   });

//   expect(mockSetBasicInfo).toHaveBeenCalledTimes(1);
//   expect(mockSetBasicInfo).toHaveBeenCalledWith({
//     ...basicInfo,
//     titlecourse: 'New Course Title',
//   });
// });

// test('handles ShortDescription textarea change', () => {
//   render(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />);

//   fireEvent.change(screen.getByPlaceholderText('Item description here...'), {
//     target: { name: 'ShortDescription', value: 'New Short Description' },
//   });

//   expect(mockSetBasicInfo).toHaveBeenCalledTimes(1);
//   expect(mockSetBasicInfo).toHaveBeenCalledWith({
//     ...basicInfo,
//     ShortDescription: 'New Short Description',
//   });
// });

// test('handles Whatlearn textarea change', () => {
//   render(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />);

//   fireEvent.change(screen.getByLabelText('What will students learn in your course?*'), {
//     target: { name: 'Whatlearn', value: 'New Learning Outcome' },
//   });

//   expect(mockSetBasicInfo).toHaveBeenCalledTimes(1);
//   expect(mockSetBasicInfo).toHaveBeenCalledWith({
//     ...basicInfo,
//     Whatlearn: 'New Learning Outcome',
//   });
// });

// test('handles courseLevel select change', async () => {
//     render(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />);
  
//     // Find the Select element by its role and name
//     const selectElement = screen.getByRole('combobox', { name: /course level/i });
    
//     // Open the Select dropdown
//     userEvent.click(selectElement);
  
//     // Wait for options to be visible
//     await waitFor(() => {
//       // Ensure options are visible in the dropdown
//       expect(screen.getByText('Beginner')).toBeInTheDocument();
//       expect(screen.getByText('Intermediate')).toBeInTheDocument();
//       expect(screen.getByText('Advanced')).toBeInTheDocument();
//     });
  
//     // Select an option
//     userEvent.click(screen.getByText('Intermediate'));
  
//     // Verify that setBasicInfo is called with the correct value
//     expect(mockSetBasicInfo).toHaveBeenCalledTimes(1);
//     expect(mockSetBasicInfo).toHaveBeenCalledWith({
//       ...basicInfo,
//       courseLevel: 'Intermediate',
//     });
//   });

// test('handles audioLanguage select change', async () => {
//   render(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />);

//   await userEvent.selectOptions(screen.getByPlaceholderText('Select audio'), ['Spanish']);

//   expect(mockSetBasicInfo).toHaveBeenCalledTimes(1);
//   expect(mockSetBasicInfo).toHaveBeenCalledWith({
//     ...basicInfo,
//     audioLanguage: 'Spanish',
//   });
// });

// test('handles closeCaption select change', async () => {
//   render(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />);

//   await userEvent.selectOptions(screen.getByPlaceholderText('Select caption'), ['French']);

//   expect(mockSetBasicInfo).toHaveBeenCalledTimes(1);
//   expect(mockSetBasicInfo).toHaveBeenCalledWith({
//     ...basicInfo,
//     closeCaption: 'French',
//   });
// });

// test('handles courseCategory select change', async () => {
//   render(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />);

//   await userEvent.selectOptions(screen.getByPlaceholderText('Web Development'), ['Data Science']);

//   expect(mockSetBasicInfo).toHaveBeenCalledTimes(1);
//   expect(mockSetBasicInfo).toHaveBeenCalledWith({
//     ...basicInfo,
//     courseCategory: 'Data Science',
//   });
// });

test('matches snapshot', () => {
  const tree = renderer
    .create(<Basic basicInfo={basicInfo} setBasicInfo={mockSetBasicInfo} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
