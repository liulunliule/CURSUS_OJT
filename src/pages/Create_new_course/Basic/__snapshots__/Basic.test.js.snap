// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="create_course_basic"
>
  <div
    className="basic_title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-face-smile basic_icon"
      data-icon="face-smile"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm177.6 62.1C192.8 334.5 218.8 352 256 352s63.2-17.5 78.4-33.9c9-9.7 24.2-10.4 33.9-1.4s10.4 24.2 1.4 33.9c-22 23.8-60 49.4-113.6 49.4s-91.7-25.5-113.6-49.4c-9-9.7-8.4-24.9 1.4-33.9s24.9-8.4 33.9 1.4zM144.4 208a32 32 0 1 1 64 0 32 32 0 1 1 -64 0zm192-32a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Basic information
  </div>
  <div
    className="basic_main_content"
  >
    <div
      className="basic_main_course-title"
    >
      <label
        htmlFor="titlecourse"
      >
        Course title*
      </label>
      <form
        action=""
        className="course_title-input"
      >
        <input
          name="titlecourse"
          onChange={[Function]}
          placeholder="Course title here"
          type="text"
          value=""
        />
        <span>
          60
        </span>
      </form>
      <p
        className="course_title-desc"
      >
        (Please make this a maximum of 100 characters and unique.)
      </p>
    </div>
    <div
      className="basic_main_short-description"
    >
      <label
        htmlFor="ShortDescription"
      >
        Short Description*
      </label>
      <form
        action=""
        className="short-description-input"
      >
        <textarea
          name="ShortDescription"
          onChange={[Function]}
          placeholder="Item description here..."
          value=""
        />
      </form>
      <p
        className="course_title-desc"
      >
        220 words
      </p>
    </div>
    <div
      className="basic_main_course-description"
    >
      <label
        htmlFor="Description"
      >
        Course Description*
      </label>
      <textarea
        onChange={[Function]}
        style={
          Object {
            "height": "150px",
          }
        }
        theme="snow"
        value=""
      />
    </div>
    <div
      className="basic_main_requirement"
    >
      <div
        className="requirement_left"
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <label
          htmlFor="Whatlearn"
        >
          What will students learn in your course?*
        </label>
        <textarea
          className="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
          disabled={false}
          name="Whatlearn"
          onBlur={[Function]}
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          rows={5}
          style={
            Object {
              "resize": undefined,
            }
          }
          value=""
        />
        <p
          className="course_title-desc"
        >
          Student will gain this skills, knowledge after completing this course. (One per line).
        </p>
      </div>
      <div
        className="requirement_right"
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <label
          htmlFor="requirement"
        >
          Requirement*
        </label>
        <textarea
          className="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          rows={5}
          style={
            Object {
              "resize": undefined,
            }
          }
          value=""
        />
        <p
          className="course_title-desc"
        >
          What knowledge, technology, tools required by users to start this course. (One per line).
        </p>
      </div>
    </div>
    <div
      className="basic_main_select"
    >
      <div
        className="basic_main_select-item"
      >
        <label
          htmlFor="courseLevel"
        >
          Course Level*
        </label>
        <div
          className="ant-select ant-select-outlined css-dev-only-do-not-override-1uq9j6g ant-select-single ant-select-show-arrow"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "height": "40px",
              "width": "100%",
            }
          }
        >
          <div
            className="ant-select-selector"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="ant-select-selection-search"
            >
              <input
                aria-autocomplete="list"
                aria-controls="undefined_list"
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-owns="undefined_list"
                autoComplete="off"
                className="ant-select-selection-search-input"
                disabled={false}
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onPaste={[Function]}
                readOnly={true}
                role="combobox"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              className="ant-select-selection-placeholder"
            >
              Nothing selected
            </span>
          </div>
          <span
            aria-hidden={true}
            className="ant-select-arrow"
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <div
        className="basic_main_select-item"
      >
        <label
          htmlFor="audioLanguage"
        >
          Audio Language*
        </label>
        <div
          className="ant-select ant-select-outlined css-dev-only-do-not-override-1uq9j6g ant-select-single ant-select-show-arrow"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "height": "40px",
              "width": "100%",
            }
          }
        >
          <div
            className="ant-select-selector"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="ant-select-selection-search"
            >
              <input
                aria-autocomplete="list"
                aria-controls="undefined_list"
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-owns="undefined_list"
                autoComplete="off"
                className="ant-select-selection-search-input"
                disabled={false}
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onPaste={[Function]}
                readOnly={true}
                role="combobox"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              className="ant-select-selection-placeholder"
            >
              Select audio
            </span>
          </div>
          <span
            aria-hidden={true}
            className="ant-select-arrow"
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <div
        className="basic_main_select-item"
      >
        <label
          htmlFor="closeCaption"
        >
          Close Caption*
        </label>
        <div
          className="ant-select ant-select-outlined css-dev-only-do-not-override-1uq9j6g ant-select-single ant-select-show-arrow"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "height": "40px",
              "width": "100%",
            }
          }
        >
          <div
            className="ant-select-selector"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="ant-select-selection-search"
            >
              <input
                aria-autocomplete="list"
                aria-controls="undefined_list"
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-owns="undefined_list"
                autoComplete="off"
                className="ant-select-selection-search-input"
                disabled={false}
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onPaste={[Function]}
                readOnly={true}
                role="combobox"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              className="ant-select-selection-placeholder"
            >
              Select caption
            </span>
          </div>
          <span
            aria-hidden={true}
            className="ant-select-arrow"
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <div
        className="basic_main_select-item"
      >
        <label
          htmlFor="courseCategory"
        >
          Course Category*
        </label>
        <div
          className="ant-select ant-select-outlined css-dev-only-do-not-override-1uq9j6g ant-select-single ant-select-show-arrow"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "height": "40px",
              "width": "100%",
            }
          }
        >
          <div
            className="ant-select-selector"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="ant-select-selection-search"
            >
              <input
                aria-autocomplete="list"
                aria-controls="undefined_list"
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-owns="undefined_list"
                autoComplete="off"
                className="ant-select-selection-search-input"
                disabled={false}
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onPaste={[Function]}
                readOnly={true}
                role="combobox"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              className="ant-select-selection-item"
              title="Web Development"
            >
              Web Development
            </span>
          </div>
          <span
            aria-hidden={true}
            className="ant-select-arrow"
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
